{"id": "49503", "title": "Connectors do not bind to their ports in Catalina.load(), but only later in start()", "description": "Testing with trunk at revision 957942, it is observed that the connectors in Catalina.load() do not bind to their ports immediately, but only later in start(). This issue may also affect version 7.0.0. In Tomcat 6, there is a separate prestartup phase which allows the connectors to bind to their ports before other parts of Tomcat are started. This is implemented through the Bootstrapload(String[]) and Bootstartstart() methods. This issue seems to break the fix for BZ 48625. To reproduce the issue, start the first instance successfully.", "OB": "In StandardServiceinitInternal(), when the connector.init() method is called, it should fail because it cannot bind to the respective port. The exception from connector.init() should be caught and if the EXITONINITFAILURE flag is set, it should be rethrown. This rethrown exception should be caught in Catalinaload() and then rethrown as an Error, which will abort the Tomcat startup.", "EB": "I expect that when the connector.init() method is called in StandardServiceinitInternal(), it should fail and throw an exception because it cannot bind to the port. This exception should be caught and, if the EXITONINITFAILURE flag is set, it should be rethrown. After that, the exception should be caught in Catalinaload() and rethrown as an Error to abort the Tomcat startup.", "SR": "Modify the StandardServiceinitInternal() method to properly handle the exception thrown by connector.init() when it fails to bind to the specified port. If the EXITONINITFAILURE flag is set, rethrow the exception. Update Catalinaload() to catch the rethrown exception and throw it as an Error to stop the Tomcat startup process."}